File: problems/2390/Lc2390.ts
function removeStars(s: string): string {
  const arr: string[] = [];
  for (const ch of s) {
    if (ch === '*') {
      arr.pop();
    } else {
      arr.push(ch);
    }
  }
  return arr.join('');
};

function main() {
  let s: string, result: string, expected: string;

  s = 'leet**cod*e'
  result = removeStars(s)
  expected = 'lecoe'
  expected === result || console.trace(`expected ${expected}, but ${result}`);

  s = 'erase*****'
  result = removeStars(s)
  expected = ''
  expected === result || console.trace(`expected ${expected}, but ${result}`);
};
main();

File: problems/2390/Lc2390.py
class Solution:
    def removeStars(self, s: str) -> str:
        lst = []
        for w in s:
            if w == '*':
                lst.pop()
            else:
                lst.append(w)
        return ''.join(lst)


if __name__ == '__main__':
    sol = Solution()
    s = 'leet**cod*e'
    result = sol.removeStars(s)
    expected = 'lecoe'
    assert expected == result, f'expected {expected}, but {result}'

    s = 'erase*****'
    result = sol.removeStars(s)
    expected = ''
    assert expected == result, f'expected {expected}, but {result}'
